op-node/service.go

```diff
@@ -290,6 +299,54 @@ func applyOverrides(ctx *cli.Context, rollupConfig *rollup.Config) {
 	}
 }
 
+func NewL1ChainConfig(chainId *big.Int, ctx *cli.Context, log log.Logger) (*params.ChainConfig, error) {
+	if chainId == nil {
+		panic("l1 chain id is nil")
+	}
+	switch {
+	case chainId.Cmp(params.MainnetChainConfig.ChainID) == 0:
+		if ctx.IsSet(flags.L1ChainConfig.Name) {
+			log.Warn("L1 chain config is set, but it is not necessary for mainnet")
+		}
+		return params.MainnetChainConfig, nil
+	case chainId.Cmp(params.SepoliaChainConfig.ChainID) == 0:
+		if ctx.IsSet(flags.L1ChainConfig.Name) {
+			log.Warn("L1 chain config is set, but it is not necessary for sepolia")
+		}
+		return params.SepoliaChainConfig, nil
+	default:
+		cf, err := NewL1ChainConfigFromCLI(log, ctx)
+		if err != nil {
+			return nil, err
+		}
+		if cf.BlobScheduleConfig == nil {
+			return nil, fmt.Errorf("L1 chain config does not have a blob schedule config")
+		}
+		return cf, nil
+	}
+}
+
+func NewL1ChainConfigFromCLI(log log.Logger, ctx *cli.Context) (*params.ChainConfig, error) {
+	l1ChainConfigPath := ctx.String(flags.L1ChainConfig.Name)
+	file, err := os.Open(l1ChainConfigPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read chain spec: %w", err)
+	}
+	defer file.Close()
```

Comment:
nit: Should we handle the error ([ref](https://gobyexample.com/defer))? A grep for `defer .*\.Close` reveals we don't handle this in other places.